### PR TITLE
Curl fail on error so we get explicit messages in case we hit github rate limit (or else)

### PR DESCRIPTION
**Pull request by Rackover**

### CODE REVIEW

File: src/common/utils/http.cpp [hunk 1]
```diff
@@ -78,10 +78,21 @@ namespace utils::http
...
+			long http_code = 0;
+			curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code);
+
+			// But the server could still report 301 or 302 which would not be good for us. 
+			if (http_code >= 200 && http_code < 300) 
+			{
+				return { std::move(buffer) };
+			}
+
+			throw std::exception(("Bad status code "+std::to_string(http_code)+" met while trying to download file "+url+"!").c_str());
```
Comment (diamante0018) [inline src/common/utils/http.cpp:95]: Uh, aren't you looking for std::runtime_error ?
You are using a non-standard implementation for std::exception as there is no constructor according to the good old C++ standard that takes a "const char*" as parameter.
https://en.cppreference.com/w/cpp/error/exception/exception

**Review by diamante0018 — CHANGES_REQUESTED** [review]
It's good you made such an addition to the code. Please see my comment

File: src/common/utils/http.cpp [hunk 2]
```diff
@@ -78,10 +78,21 @@ namespace utils::http
...
 		{
-			return {std::move(buffer)};
+			long http_code = 0;
+			curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code);
+
+			// But the server could still report 301 or 302 which would not be good for us. 
+			if (http_code >= 200 && http_code < 300) 
+			{
+				return { std::move(buffer) };
+			}
```
Comment (momo5502) [inline src/common/utils/http.cpp:93]: You shouldn't get any 302 or 301, cause of CURLOPT_FOLLOWLOCATION

File: src/common/utils/http.cpp (hunk 1 above)
Comment (momo5502) [reply src/common/utils/http.cpp:95]: Yup, std::runtime_error is needed here

**Review by momo5502 — CHANGES_REQUESTED** [review]

**Comment by Rackover** [comment]
I think it's ready. the >200 check is a bit extra but it should work

**Review by momo5502 — APPROVED** [review]

**Pull request merged**